### PR TITLE
chore: Render tailored html files in some cases

### PR DESCRIPTION
**Pull request by sharat87**
When opening Appsmith in edit mode, we _try_ to render the specialized `edit.html`, and for opening in view mode, we _try_ to render the specialized `view.html` files, instead of the `index.html`. If those files aren't available, we fallback to `index.html`, so there's no disruption in service.

These specialized files are, today, identical to `index.html`, but contain a few extra preloads that are tuned to work better for those pages.

Learn more at https://github.com/appsmithorg/appsmith/pull/21933.

### CODE REVIEW

File: deploy/docker/templates/nginx/nginx-app-http.conf.template.sh [hunk 1]
```diff
@@ -73,6 +73,14 @@ server {
     try_files \$uri /index.html =404;
   }
 
+  location ~ ^/app/[^/]+/[^/]+/edit\b {
```
Comment (SatishGandham) [inline deploy/docker/templates/nginx/nginx-app-http.conf.template.sh:76]: I don't know the exact regex but, if there is something after the edit, then it is preceded by a `/`
This regex matches `/app/hello/123/editfoo`, it is not a problem now, but there may be a case in future.

**Review by SatishGandham — CHANGES_REQUESTED** [review]

**Review by iamakulov — APPROVED** [review]
This will work for Cypress tests as well, right? (We have a Cypress test that verifies that the preloads in `view.html`/`index.html` weren’t broken.)

**Comment by sharat87** [comment]
@iamakulov, yes. Cypress tests should be able to verify if things continue to work as is with the changes here.

**Review by SatishGandham — APPROVED** [review]

**Comment by sharat87** [comment]
Just checking in, we still want/need this, right? Anything I can do to help move this faster?

**Comment by sharat87** [comment]
Changes are already in `release`.

**Pull request closed without merging**